test/main/transaction/index.test.js

```diff
@@ -485,3 +493,149 @@ describe('#sign', () => {
+    const request = Request({
+      from,
+      data: '0x',
+      value: '0x01'
+    })
+
+    expect(classifyTransaction(request)).toBe(TxClassification.CONTRACT_DEPLOY)
+  })
+  it('Correctly classifies contract deployments with no data or value', () => {
+    const request = Request({
+      from,
+      data: '0x',
+      value: '0x'
+    })
+
+    expect(classifyTransaction(request)).toBe(TxClassification.CONTRACT_DEPLOY)
+  })
+
+  // Sending Data
+  it('Correctly classifies sending data to an EOA', () => {
+    const request = Request({
+      from,
+      to,
+      data: '0x6080604052'
+    })
+
+    expect(classifyTransaction(request, 'external')).toBe(TxClassification.SEND_DATA)
+  })
+  it('Correctly classifies sending data to an EOA with value', () => {
+    const request = Request({
+      from,
+      to,
+      data: '0x6080604052',
+      value: '0x01'
+    })
+
+    expect(classifyTransaction(request, 'external')).toBe(TxClassification.SEND_DATA)
+  })
+
+  it('Correctly classifies sending data to an unknown recipientType', () => {
+    const request = Request({
+      from,
+      to,
+      data: '0x6080604052'
+    })
+
+    expect(classifyTransaction(request)).toBe(TxClassification.CONTRACT_CALL)
+  })
+  it('Correctly classifies sending data to an unknown recipientType and a value', () => {
+    const request = Request({
+      from,
+      to,
+      data: '0x6080604052',
+      value: '0x01'
+    })
+
+    expect(classifyTransaction(request)).toBe(TxClassification.CONTRACT_CALL)
+  })
+
+  //Contract Calls
+  it('Correctly classifies calling a contract function', () => {
+    const request = Request({
+      from,
+      to,
+      data: '0x6080604052'
+    })
+
+    expect(classifyTransaction(request, 'contract')).toBe(TxClassification.CONTRACT_CALL)
+  })
+  it('Correctly classifies calling a contract function with a value', () => {
+    const request = Request({
+      from,
+      to,
+      data: '0x6080604052',
+      value: '0x01'
+    })
+
+    expect(classifyTransaction(request, 'contract')).toBe(TxClassification.CONTRACT_CALL)
+  })
+
+  it('Correctly classifies a native transfer to an EOA', () => {
+    const request = Request({
+      from,
+      to,
+      value: '0x01'
+    })
+
+    expect(classifyTransaction(request, 'external')).toBe(TxClassification.NATIVE_TRANSFER)
+  })
+  it('Correctly classifies a native transfer to an EOA with no value', () => {
+    const request = Request({
+      from,
+      to,
+      value: '0x0'
+    })
+
+    expect(classifyTransaction(request, 'external')).toBe(TxClassification.NATIVE_TRANSFER)
+  })
+  it('Correctly classifies sending a native transfer to a contract', () => {
+    const request = Request({
+      from,
+      to,
+      value: '0x01'
+    })
+
+    expect(classifyTransaction(request, 'contract')).toBe(TxClassification.NATIVE_TRANSFER)
+  })
+  it('Correctly classifies sending a native transfer to a contract with no value', () => {
```

Comment:
formatting nit - we don't have a standard way of writing these (I personally like `it('should ...'` though can appreciate the more succinct approach).  However, I do think we should standardise on a lowercase first letter.